apps/dashboard/app/javascript/icon_picker.js

```diff
@@ -64,20 +64,52 @@ function addSearch(){
   });
 }
 
-function searchIcons(event){
-  const searchString = event.target.value;
-  const rex = new RegExp(searchString, "g");
-  hiddenIcons = true;
+/**
+ * Creates an inverted index of ALL_ICONS.
+ * Returns an array type.
+ **/
+ function createInvertedIndex(arr) {
+  const invertedIndex = {};
+
+  arr.forEach((str, index) => {
+    for (let i = 0; i < str.length; i++) {
+      const char = str[i];
+      if (!invertedIndex[char]) {
+        invertedIndex[char] = [];
+      }
+      invertedIndex[char].push(index);
+    }
+  });
 
-  ALL_ICONS.forEach(name => {
+  return invertedIndex;
+}
+ 
+let invertedIndex = createInvertedIndex(ALL_ICONS);
+
+function searchIcons(event) {
+  const searchString = event.target.value.toLowerCase(); // Convert input to lowercase for case-insensitive search
+  const iconIds = Object.keys(invertedIndex); // Get all keys (characters) from the inverted index
+  const resultIndices = new Set(); // Set to store indices of matching icons
+
+  iconIds.forEach(char => {
```

Comment:
Wouldn't it be better to search over `searchString.split('')` instead of all the `iconIds`?  I think the result is the same, but you'd search over fewer elements.

Reply:
Yea if I add a little debug logic to find how many comparisons this makes, it makes a constant `9388` comparisons, regardless of the search string.

Reply:
Circling back to this, just `searchString.split('')` can explode the number of operations because there's no filtering.

something like this (with filtering) can limit the number of operations but there is still an upper bound of `9388` when you search for something like 
`abcdefghijklmnopqrstuvwxyz260-` (the content of `iconIds`).

```diff
-  const iconIds = Object.keys(invertedIndex); // Get all keys (characters) from the inverted index
+  const indexKeys = Object.keys(invertedIndex); // Get all keys (characters) from the inverted index
   const resultIndices = new Set(); // Set to store indices of matching icons
+  const uniqueSearchCharacters = new Set(searchString.split(''));
+  const searchCharacters = [...uniqueSearchCharacters].filter((char) => indexKeys.includes(char));
 
-  iconIds.forEach(char => {
+  searchCharacters.forEach(char => {
```